src/server/pachyderm_test.go

```diff
@@ -1381,10 +1394,129 @@ func TestProvenance(t *testing.T) {
 	require.Equal(t, 2, len(commitInfos))
 	for _, commitInfo := range commitInfos {
 		// C takes the diff of 2 files that should always be the same, so we
-		// expect no output and thus no file
-		_, err := c.InspectFile(cPipeline, commitInfo.Commit.ID, "file", "", nil)
-		require.YesError(t, err)
+		// expect an empty file
+		fileInfo, err := c.InspectFile(cPipeline, commitInfo.Commit.ID, "file", "", nil)
+		require.NoError(t, err)
+		require.Equal(t, uint64(0), fileInfo.SizeBytes)
+	}
+}
+
+func TestDirectory(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+
+	c := getPachClient(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+	defer cancel() //cleanup resources
+
+	job1, err := c.PpsAPIClient.CreateJob(context.Background(), &ppsclient.CreateJobRequest{
+		Transform: &ppsclient.Transform{
+			Cmd: []string{"sh"},
+			Stdin: []string{
+				"mkdir /pfs/out/dir",
+				"echo foo >> /pfs/out/dir/file",
+			},
+		},
+		Parallelism: 3,
+	})
+	require.NoError(t, err)
+	inspectJobRequest1 := &ppsclient.InspectJobRequest{
+		Job:        job1,
+		BlockState: true,
+	}
+	jobInfo1, err := c.PpsAPIClient.InspectJob(ctx, inspectJobRequest1)
+	require.NoError(t, err)
+	require.Equal(t, ppsclient.JobState_JOB_SUCCESS, jobInfo1.State)
+
+	var buffer bytes.Buffer
+	require.NoError(t, c.GetFile(jobInfo1.OutputCommit.Repo.Name, jobInfo1.OutputCommit.ID, "dir/file", 0, 0, "", nil, &buffer))
```

Comment:
What do we expect if I try to get a file thats really a directory? I'm guessing we return an empty file? 

We should have a test for whatever behavior we support.